packages/catalog-platformstate-writer/test/consumerServiceV2.test.ts

```diff
@@ -0,0 +1,1205 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { fail } from "assert";
+import {
+  afterAll,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from "vitest";
+import {
+  Descriptor,
+  EService,
+  EServiceDescriptorActivatedV2,
+  EServiceDescriptorArchivedV2,
+  EServiceDescriptorPublishedV2,
+  EServiceDescriptorSuspendedV2,
+  EServiceEventEnvelope,
+  PlatformStatesCatalogEntry,
+  TokenGenerationStatesClientPurposeEntry,
+  descriptorState,
+  generateId,
+  itemState,
+  makeGSIPKEServiceIdDescriptorId,
+  makePlatformStatesEServiceDescriptorPK,
+  makeTokenGenerationStatesClientKidPurposePK,
+  toEServiceV2,
+} from "pagopa-interop-models";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  getMockDescriptor,
+  getMockEService,
+  getMockDocument,
+  getMockTokenStatesClientPurposeEntry,
+  buildDynamoDBTables,
+  deleteDynamoDBTables,
+} from "pagopa-interop-commons-test";
+import {
+  readCatalogEntry,
+  readTokenStateEntriesByEserviceIdAndDescriptorId,
+  writeCatalogEntry,
+} from "../src/utils.js";
+import { handleMessageV2 } from "../src/consumerServiceV2.js";
+import { config, sleep, writeTokenStateEntry } from "./utils.js";
+
+describe("integration tests V2 events", async () => {
+  if (!config) {
+    fail();
+  }
+  const dynamoDBClient = new DynamoDBClient({
+    credentials: { accessKeyId: "key", secretAccessKey: "secret" },
+    region: "eu-central-1",
+    endpoint: `http://${config.tokenGenerationReadModelDbHost}:${config.tokenGenerationReadModelDbPort}`,
+  });
+  beforeEach(async () => {
+    await buildDynamoDBTables(dynamoDBClient);
+  });
+  afterEach(async () => {
+    await deleteDynamoDBTables(dynamoDBClient);
+  });
+  const mockDate = new Date();
+  beforeAll(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(mockDate);
+  });
+  afterAll(() => {
+    vi.useRealTimers();
+  });
+
+  describe("EServiceDescriptorActivated", () => {
+    it("should do no operation if the entry already exists: incoming has version 1; previous entry has version 2", async () => {
```

Comment:
the use case is more clear (apply the following changes where required)
```suggestion
    it("should do no operation if the existing table entry is more recent", async () => {
```

Reply:
Renamed